src/main/java/com/vinylteam/vinyl/service/impl/DefaultDiscogsService.java

```diff
@@ -87,14 +88,22 @@ public String getDiscogsLink(String artist, String release, String fullName) thr
         String requestBody;
         String discogsLink = "";
         if (artist == null || release == null || fullName == null
-                || artist.isEmpty() || release.isEmpty() || fullName.isEmpty()) {
+                || artist.isEmpty() || release.isEmpty() || fullName.isEmpty()
+                || !fullName.toLowerCase().contains(artist.toLowerCase())
+                || !fullName.toLowerCase().contains(release.toLowerCase())) {
             return discogsLink;
         }
-        String artistComparing = getParametersForComparison(artist);
-        log.debug("Prepared artistComparing for comparison with release from Discogs {'artistComparing':{}}", artistComparing);
-        String releaseComparing = getParametersForComparison(release);
-        log.debug("Prepared releaseComparing for comparison with artist from Discogs {'releaseComparing':{}}", releaseComparing);
-        query = fullName.replace(" ", "+");
+        artist = artist.toLowerCase();
+        release = release.toLowerCase();
+        fullName = fullName.toLowerCase();
+        String[] preparedFullNameForMatching;
+        if (artist.toLowerCase().contains("various")) {
+            preparedFullNameForMatching = Arrays.stream(release.split("[- ()!@#$%^&*_+={}:;\"']")).filter(e -> e.trim().length() > 0).toArray(String[]::new);
+        } else {
+            preparedFullNameForMatching = Arrays.stream(fullName.split("[- ()!@#$%^&*_+={}:;\"']")).filter(e -> e.trim().length() > 0).toArray(String[]::new);
```

Comment:
The same code, starting from .split() and further... can it be declared in one place?